### ISSUE TITLE
[sim] Build Error with Robots

### ISSUE THREAD

**Issue by gulyx** [LearnPAd/learnpad#440]
@tomjorquera, compiling the current version of the platform it looks that the SIM component fail the build.
Please see in this report : http://hastebin.com/aloguyikiw.avrasm
the errors @andreapolini got on his machine. I had similar errors too but only after I fetched and merged
the latest version. Probably this is a regression due to PR #438 .

**Comment by tomjorquera** [comment]
Hi @gulyx,

This problem is actually unrelated to the robots. It was caused by a... rather mysterious change in  commit https://github.com/LearnPAd/learnpad/commit/2b90e1007d469383ac16165486b53f3b982fd637

This commit included a change that did set some previously public attributes to private in the `ProcessInstanceData` class. This obviously caused breakages in the implementation of the API. I am not clear on why this change did not cause any warning from travis (the commit was not merged from a PR as usual, but still...)

Anyways PR #452 will fix the issue.

**Comment by gulyx** [comment]
thanks @tomjorquera, actually i do not know why the visibility of that attributes have been modified. 
Btw, I leave to you the task of merging the PR ... and hopefully this will not introduce any regression :-D

**Comment by woshilapin** [comment]
I was the on who change the visibility because I'm using this object in Velocity and it seems that Velocity doesn't work with class with only public fields (crappy language).

**Comment by gulyx** [comment]
Thanks for the answer @woshilapin, where did u accessed this object exactly in the code?

So at this point, @tomjorquera, @cjdelisle, do u think that PR #452 can be merged?

**Comment by tomjorquera** [comment]
Thanks @woshilapin for chiming in!

In this case, PR #452 should not be merged. I will close it and prepare a new PR.

**Comment by tomjorquera** [comment]
PR #458 fixes the issue
